.circleci/config.yml

```diff
@@ -0,0 +1,262 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+orbs:
+  docker: circleci/docker@1.7
+
+commands:
+  setup-remote-docker-with-experimental-feats:
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+          version: 20.10.7
+      - run:
+          # We need this in order to `squash` images
+          # https://support.circleci.com/hc/en-us/articles/360050448351-Enabling-Docker-Experimental-Features-on-Remote-Docker
+          name: Enable Docker experimental features
+          command: |
+            ssh remote-docker \<<EOF
+              sudo bash -c 'echo "{\"experimental\": true}" > /etc/docker/daemon.json'
+              sudo systemctl restart docker
+            EOF
+
+  compute-docker-tags:
+    parameters:
+      image-name:
+        type: string
+      squash:
+        type: boolean
+        default: false
+    steps:
+      # Sanitize the $BRANCH name and create the utility vars with the final
+      # tags/full image names (which may or may not be squashed)
+      - run:
+          name: Sanitize branch name and create related utility vars
+          command: |
+            echo 'export BRANCH=${CIRCLE_BRANCH//\//-}' >> "$BASH_ENV"
+            echo 'export BRANCH_IMAGE_TAG=${BRANCH}<<# parameters.squash >>-squash<</ parameters.squash >>' >> "$BASH_ENV"
+            echo 'export BRANCH_IMAGE_URL=<< parameters.image-name >>:${BRANCH_IMAGE_TAG}' >> "$BASH_ENV"
+            echo 'export SHA1_IMAGE_TAG=${CIRCLE_SHA1}<<# parameters.squash >>-squash<</ parameters.squash >>' >> "$BASH_ENV"
+            echo 'export SHA1_IMAGE_URL=<< parameters.image-name >>:${SHA1_IMAGE_TAG}' >> "$BASH_ENV"
+            echo 'export IMAGE_TAG_LIST=${SHA1_IMAGE_TAG},${BRANCH_IMAGE_TAG}' >> "$BASH_ENV"
+      # Only create the GIT_TAG related tags if we're building a tag
+      - when:
+          condition: << pipeline.git.tag >>
+          steps:
+            - run:
+                name: Create git tag related image tag vars
+                command: |
+                  echo 'export GIT_TAG_IMAGE_TAG=${CIRCLE_TAG}<<# parameters.squash >>-squash<</ parameters.squash >>' >> "$BASH_ENV"
+                  echo 'export GIT_TAG_IMAGE_URL=<< parameters.image-name >>:${GIT_TAG_IMAGE_TAG}' >> "$BASH_ENV"
+                  echo 'export IMAGE_TAG_LIST=${SHA1_IMAGE_TAG},${BRANCH_IMAGE_TAG},${GIT_TAG_IMAGE_TAG}' >> "$BASH_ENV"
```

Comment:
We can't compute `environment` variables via other environment variables in CCI via the regular `environment` entry. This is the suggested approach to handle this problem - https://circleci.com/docs/2.0/env-vars/#using-parameters-and-bash-environment